sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -155,9 +155,15 @@ def rank_error(msg):
               or str(response.currentState) == 'JOB_STATE_UPDATED'
               or str(response.currentState) == 'JOB_STATE_DRAINED'):
             break
-          # The job has failed; ensure we see any final error messages.
-          sleep_secs = 1.0      # poll faster during the final countdown
-          final_countdown_timer_secs -= sleep_secs
+
+          # Check that job is in a post-preparation state before starting the
+          # final countdown.
+          if (str(response.currentState) != 'JOB_STATE_PENDING'
+              and str(response.currentState) != 'JOB_STATE_QUEUED'):
```

Comment:
How about `if str(response.currentState) not in ('JOB_STATE_PENDING', 'JOB_STATE_QUEUED'):`?